### ISSUE TITLE
Make it easier to highlight description/label for filter on type in tree

### ISSUE THREAD

**Issue by roblourens** [Microsoft/vscode#67080]
Some tree rows that show files have a 'label' part for the filename and a 'description' part for the rest of the path. We should be able to search on both of these and highlight them separately but currently it's not easy.

Currently each row item has one string from `getKeyboardNavigationLabel` and the tree does the fuzzy matching and gives the renderer one `filterData`. The renderer wants to pass separate `matches` and `descriptionMatches` t the ResourceLabel but it's not easy to translate the one filterData into two parts.

It would be easier if 
- The row could return multiple keyboard navigation labels, and the renderer would then get multiple FuzzyScores
- Or if the `keyboardNavigationLabelProvider` did the fuzzy matching itself and returns some result object that the renderer gets

Thoughts?

**Comment by joaomoreno** [comment]
Yeah that's the reason I was reluctant in marrying `getKeyboardNavigationLabel` with this feature. Multiple labels might be an interesting way of doing this. But if keyboard navigation config is set to `simple`, only the first would matter. Makes sense?

**Comment by roblourens** [comment]
Yeah that makes sense.

**Comment by roblourens** [comment]
The specific inspiration for this was the search tree. But it's not a high priority thing.
